fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/CreateResourceServiceImpl.java

```diff
@@ -236,9 +237,14 @@ private void checkParent(final String txId, final PersistentStorageSession pSess
      * @return a list of LINK headers with rel="type"
      */
     private List<String> getTypes(final List<String> headers) {
-        return getLinkHeaders(headers) == null ? emptyList() : getLinkHeaders(headers).stream()
-                .filter(p -> p.getRel().equalsIgnoreCase("type")).map(Link::getUri)
-                .map(URI::toString).collect(Collectors.toList());
+        final List hdrobjs = getLinkHeaders(headers);
+        try {
+            return hdrobjs == null ? emptyList() : getLinkHeaders(headers).stream()
+                    .filter(p -> p.getRel().equalsIgnoreCase("type")).map(Link::getUri)
+                    .map(URI::toString).collect(Collectors.toList());
+        } catch ( Exception e ) {
+            throw new BadRequestException("Invalid type found");
```

Comment:
Can you change this to `new BadRequestException("Invalid type found", e)` so that the stack trace is not lost.

Reply:
..or maybe to be even more clear:
```
new BadRequestException("Invalid Link header type found", e)
```